base/MAPL_EASEConv.F90

```diff
@@ -0,0 +1,818 @@
+
+    ! ---------------------------------------------------------------------
+
+    real*8                     :: map_scale_m, CELL_km, r0, s0, Rg
+    real                       :: tmplon
+    character(3)               :: grid
+    
+    if (     index(EASELabel,'M36') /=0 ) then
+       grid='M36'
+    else if (index(EASELabel,'M25') /=0 ) then
+       grid='M25'
+    else if (index(EASELabel,'M09') /=0 ) then
+       grid='M09'
+    else if (index(EASELabel,'M03') /=0 ) then
+       grid='M03'
+    else if (index(EASELabel,'M01') /=0 ) then
+       grid='M01'
+    else
+       print*,"ease_extent(): unknown grid projection and resolution: "//trim(EASELabel)//"  STOPPING."
+       stop
+    endif
+    
+    if(     index(EASELabel,'EASEv2') /=0) then
+
+       call easeV2_get_params(grid, map_scale_m, cols, rows, r0, s0)
+
+       if(present(cell_area)) cell_area = map_scale_m**2
+
+    else if(index(EASELabel,'EASEv1') /=0) then
+
+       call easeV1_get_params(grid, CELL_km, cols, rows, r0, s0, Rg)
+
+       if(present(cell_area)) cell_area = CELL_km**2 * 1000. * 1000.
+
+    else
+
+       print*,"ease_extent(): unknown grid version: "//trim(EASELabel)//"  STOPPING."
+       stop
+
+    endif
+
+    ! get lat/lon of corner grid cells
+    ! 
+    ! recall that EASE grid indexing is zero-based
+
+    if (present(ll_lat))  call ease_inverse(EASElabel, 0., rows-0.5, ll_lat, tmplon)     
+    if (present(ur_lat))  call ease_inverse(EASElabel, cols-1.0,     -0.5, ur_lat, tmplon)     
+    
+    if (present(ll_lon))  ll_lon = -180.
+    if (present(ur_lon))  ur_lon =  180.
+
+  end subroutine ease_extent
+
+  ! *******************************************************************
+  !
+  !   EASEv1 routines (private)
+  !
+  ! *******************************************************************
+  
+  subroutine easeV1_convert (grid, lat, lon, r, s)
+    
+    ! convert geographic coordinates (spherical earth) to 
+    ! azimuthal equal area or equal area cylindrical grid coordinates
+    ! 
+    ! status = easeV1_convert (grid, lat, lon, r, s)
+    ! 
+    ! input : grid - projection name '[M][xx]'
+    !            where xx = approximate resolution [km]
+    !               ie xx = "01", "03", "09", "36"       (SMAP)
+    !               or xx = "12", "25"                   (SSM/I, AMSR-E)
+    ! 	    lat, lon = geo. coords. (decimal degrees)
+    ! 
+    ! output: r, s - column, row coordinates
+    ! 
+    ! result: status = 0 indicates normal successful completion
+    ! 		-1 indicates error status (point not on grid)
+    ! 
+    ! --------------------------------------------------------------------------
+        
+    character*(*), intent(in)  :: grid
+    real,          intent(in)  :: lat, lon
+    real,          intent(out) :: r, s
+
+    ! local variables
+    
+    integer :: cols, rows
+    real*8  :: Rg, phi, lam, rho, CELL_km, r0, s0
+    
+    real*8, parameter :: PI = easeV1_PI
```

Comment:
How many values of pi does mapl need?

Reply:
I have changed to MAPL's PI and all the REAL64  parameters' definition.  However,  I believe the change will make it non-zero diff, especially in BCS. I may have to change it back to the 'wrong' way for zero-diff.

Reply:
For the parameters, we have found that with optimization on, there usually is no difference.    The compiler apparently sees the advantage of not truncating just to promote.  But ... no guarantees.

For PI - if there is a difference that would be bad.   I hope not.

If this is a problem for @gmao-rreichle, we can relax this requirement for this PR and just add it to the list of things to fix in MAPL3.

Reply:
@gmao-rreichle For PI - yes I don't really have an argument other than it is what it is.   But for the other parameters, I suspect you are getting different values with debug vs non-debug.   I've not looked closely though.  If all of the literals are real integers (e.g., 123.0) then it may not matter.    But ... in the past, MAPL had things like:
```
real(kind=REAL64), parameter :: pi = 3.14159265...
```
And the value depended on optimization until we fixed it as:
```
real(kind=REAL64), parameter :: pi = 3.14159265...d+0
```